ch12.md

```diff
@@ -0,0 +1,256 @@
+
+And now for our next trick, we'll look at traversals. We'll watch types soar over one another as if they were trapeze artists holding our value in tact. We'll reorder effects like the trolleys in a tilt-a-whirl. When our containers get intertwined like the limbs of a contortionist, we can use this interface to straighten things out. We'll witness different effects with different orderings. Fetch me my pantaloons and slide whistle, let's get started.
+
+## Types n' types
+
+Let's get weird:
+
+```js
+// readFile :: FileName -> Task Error String
+
+// firstWords :: String -> String
+const firstWords = compose(join(' '), take(3), split(' '))
+
+// tldr :: FileName -> Task Error String
+const tldr = compose(map(firstWords), readFile)
+
+map(tldr, files)
+// [Task('hail the monarchy'), Task('smash the patriarchy')]
+```
+
+Here we read a bunch of files and end up with a useless array of tasks. How might we fork each one of these? It would be most agreeable if we could switch the types around to have `Task Error [String]` instead of `[Task Error String]`. That way, we'd have one future value holding all the results, which is much more amendable to our async needs than several future values arriving at their leisure.
+
+Here's one last example of a sticky situation:
+
+```js
+// getAttribute :: String -> Node -> Maybe String
+// $ :: Selector -> IO Node
+
+// getControlNode :: IO (Maybe (IO Node))
+const getControlNode = compose(map(map($)), map(getAttribute('aria-controls')), $)
+```
+
+Look at those `IO`s longing to be together. It'd be just lovely to `join` them, let them dance cheek to cheek, but alas a `Maybe` stands between them like a chaperone at prom. Our best move here would be to shift their positions next to one another, that way each type can be together at last and our signature can be simplified to `IO (Maybe Node)`.
+
+## Type Feng Shui
+
+The *Traversable* interface consists of two glorious functions: `sequence` and `traverse`.
+
+Let's rearrange our types using `sequence`:
+
+```js
+sequence(Array, Maybe(['the facts'])) // [Maybe('the facts')]
+sequence(Task.of, Map({a: Task.of(1), b: Task.of(2)})) // Task(Map({a: 1, b: 2}))
+sequence(IO.of, Right(IO.of('buckle my shoe'))) // IO(Right('buckle my shoe'))
+sequence(Either.of, [Right('wing')]) // Right(['wing'])
+sequence(Task.of, Left('wing')) // Task(Left('wing'))
+```
+
+See what has happened here? Our nested type gets turned inside out like a pair of leather trousers on a humid summer night. The inner functor is shifted to the outside and vice versa. It should be known that `sequence` is bit particular about its arguments. It looks like this:
+
+```js
+// sequence :: (Traversable t, Applicative f) => (a -> f a) -> t (f a) -> f (t a)
+const sequence = (point, x) => x.sequence(point)
```

Comment:
So glad you went with the pointed function over `TypeRep` of an `Applicative`. I find this interface a lot more handy and natural.

Reply:
:) Thanks for looking this over, btw!